src/components/Payroll/PayrollConfiguration/PayrollConfigurationPresentation.tsx

```diff
@@ -1,90 +1,181 @@
+import type { EmployeeCompensations } from '@gusto/embedded-api/models/components/payrollshow'
+import type { Employee } from '@gusto/embedded-api/models/components/employee'
+import type { PayrollPayPeriodType } from '@gusto/embedded-api/models/components/payrollpayperiodtype'
+import { useTranslation } from 'react-i18next'
+import {
+  getPayRateDisplay,
+  getRegularHours,
+  getTotalPtoHours,
+  getAdditionalEarnings,
+  getReimbursements,
+} from '../helpers'
 import { DataView, Flex } from '@/components/Common'
 import { useComponentContext } from '@/contexts/ComponentAdapter/useComponentContext'
 import { HamburgerMenu } from '@/components/Common/HamburgerMenu'
 import PencilSvg from '@/assets/icons/pencil.svg?react'
-
-//TODO: Use Speakeasy type
-interface Employee {
-  employeeId: string
-}
+import { formatNumberAsCurrency } from '@/helpers/formattedStrings'
+import { parseDateStringToLocal } from '@/helpers/dateFormatting'
+import { useLocale } from '@/contexts/LocaleProvider/useLocale'
 
 interface PayrollConfigurationPresentationProps {
-  employees: Employee[]
+  employeeCompensations: EmployeeCompensations[]
+  employeeDetails: Employee[]
+  payPeriod?: PayrollPayPeriodType
   onBack: () => void
   onCalculatePayroll: () => void
-  onEdit: (employee: Employee) => void
+  onEdit: (employee: { employeeId: string }) => void
 }
 
 export const PayrollConfigurationPresentation = ({
-  employees,
+  employeeCompensations,
+  employeeDetails,
+  payPeriod,
   onBack,
   onEdit,
   onCalculatePayroll,
 }: PayrollConfigurationPresentationProps) => {
-  const { Alert, Button, Heading, Text } = useComponentContext()
+  const { Alert, Button, Heading, Text, Badge } = useComponentContext()
+  const { t } = useTranslation('Payroll.PayrollConfiguration')
+  const { locale } = useLocale()
+
+  const employeeMap = new Map(employeeDetails.map(employee => [employee.uuid, employee]))
+
+  const getPageTitle = () => {
```

Comment:
Could probably live outside of render function scope since it doesn't depend on rendered details. Minor thing but wouldn't need to be recreated on rerender